### PR TITLE
feat(support): Move SecureValuesPreprocessor to @appum/logger

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
BREAKING CHANGE: Removed the 'log-internals' module from '@appium/support' 
BREAKING CHANGE: Removed the 'loadSecureValuesPreprocessingRules' API from '@appium/support -> logging' 

The SecureValuesPreprocessor has been moved directly to the `@appum/logger` module and set as a private logger instance property. 
The `loadSecureValuesPreprocessingRules` method is now public instance method of the Logger class. Unit tests were ported to the `@appium/logger` module as well.

### CODE REVIEW

**Review by KazuCocoa — APPROVED** [review]
lg.

This GitHub's squash and merge commit message must not include breaking change messages since this has `appium` package modification.

**Comment by eglitise** [comment]
Like Kazu said, this would bump Appium to 3.0, so I would suggest separating the change in the appium package into its own PR.

File: packages/logger/lib/log.ts [hunk 1]
```diff
@@ -1,13 +1,22 @@
...
+  StyleObject,
+  Logger,
+  LogLevel,
+  PreprocessingRulesLoadResult,
+  LogFiltersConfig
+} from './types';
 import type {Writable} from 'node:stream';
 import {AsyncLocalStorage} from 'node:async_hooks';
 import { unleakString } from './utils';
+import { SecureValuesPreprocessor } from './secure-values-prepreocessor';
```
Comment (eglitise) [inline packages/logger/lib/log.ts:19]: Typo in file name

Comment (mykola-mokhnach) [reply packages/logger/lib/log.ts:19]: thanks. good catch

**Pull request merged**